### PR TITLE
Add cross-cluster search remote cluster info API

### LINKED ISSUE #23925: Cross-cluster search remote cluster status API
With cross-cluster search configured, Kibana needs to be able to:

* discover the names of the remote clusters
* discover the remote cluster's `http.publish_address`
* check if remote clusters are contactable

This is made more complicated by the fact that cluster settings can be `persistent`, `transient`, or `default` (configured in the `elasticsearch.yml` file).

Let's add a remote cluster status API like the following:

    GET _remote/status

which could return something like:

```
{
  "cluster_one": {
    "seeds": [
      "10.0.0.1:9300"
    ],
    "http_addresses": [
      "10.0.0.1:9200",
      "10.0.0.1:9201",
      "10.0.0.1:9202"
    ],
    "connected": true,
    "latency_in_ms": 4,
    "connections_per_cluster": 3,
    "initial_connect_timeout": "30s",
    "level": "cluster|node"
  },
  "cluster_two": {...}
}
```

The `http_addresses` could be pre-configured (see https://github.com/elastic/elasticsearch/issues/23924) or be sniffed from the remote cluster's node info.  (the same questions about HTTPS apply).

The `level` key would indicate whether this remote cluster has been configured via the cluster settings or locally on the node.

### PR DESCRIPTION
**Pull request by s1monw**
This commit adds an API to discover information like seed nodes,
http addresses and connection status of a configured remote cluster.

Closes #23925

### CODE REVIEW

**Review by clintongormley — COMMENTED** [review]
The REST tests look good to me.  Only thing missing is the docs

**Comment by s1monw** [comment]
@clintongormley thanks for taking a look. I will add docs based on what I have. I really wanted to get a first impression before I do that..

**Comment by s1monw** [comment]
@clintongormley I added docs, can you look again?

**Comment by s1monw** [comment]
@javanna can you do a review of the java parts here please

File: docs/reference/cluster/remote-info.asciidoc [hunk 1]
```diff
@@ -0,0 +1,36 @@
+[[cluster-remote-info]]
+== Remote Cluster Info
+
+The cluster remote info API allows to retrieve all of the configured
+remote cluster information.
+
+[source,js]
+--------------------------------------------------
+curl -XGET 'http://localhost:9200/_remote/info'
```
Comment (clintongormley) [inline docs/reference/cluster/remote-info.asciidoc]: Use console format:

    [source,js]
    ----------------------------------
    GET /_remote/info
    ----------------------------------
    // CONSOLE

**Review by clintongormley — APPROVED** [review]
LGTM other than the fact that the request example should use CONSOLE format.

Would be good to include an actual docs test there if possible (local cluster?)

**Comment by s1monw** [comment]
> Would be good to include an actual docs test there if possible (local cluster?)

I am hesitating to do that. I think we shouldn't do it since I want to remove the ability to self-reference a cluster as a remote.

**Review by javanna — APPROVED** [review]
LGTM

**Pull request merged**